src/libs/queries/extApi/simulator.ts

```diff
@@ -0,0 +1,107 @@
+import { useQuery } from '@tanstack/react-query';
+import { QueryKey } from 'libs/queries/queryKey';
+import { SimulatorResultSearch } from 'libs/routing';
+import { carbonApi } from 'utils/carbonApi';
+import { FIVE_MIN_IN_MS } from 'utils/time';
+
+export interface SimulatorResult {
+  CASH: {
+    balance: string[];
+    fee: string[];
+  };
+  RISK: {
+    balance: string[];
+    fee: string[];
+  };
+  dates: number[];
+  min_bid: string;
+  max_bid: string;
+  min_ask: string;
+  max_ask: string;
+  bid: string[];
+  ask: string[];
+  hodl_value: string[];
+  portfolio_cash: string[];
+  portfolio_risk: string[];
+  portfolio_value: string[];
+  portfolio_over_hodl: string[];
+  prices: string[];
+}
+
+export type SimulatorData = {
+  date: number;
+  price: number;
+  ask: number;
+  bid: number;
+  portfolioOverHodl: number;
+  portfolioValue: number;
+  hodlValue: number;
+  balanceCASH: number;
+  portionCASH: number;
+  balanceRISK: number;
+  portionRISK: number;
+};
+
+export type SimulatorBounds = {
+  askMin: number;
+  askMax: number;
+  bidMin: number;
+  bidMax: number;
+};
+
+export type SimulatorReturn = {
+  data: Array<SimulatorData>;
+  bounds: SimulatorBounds;
+  roi?: number;
+  gains?: number;
+};
+
+export const useGetSimulator = (params: SimulatorResultSearch) => {
+  return useQuery<SimulatorReturn, Error>(
+    QueryKey.simulator(params),
+    async () => {
+      try {
+        const res = await carbonApi.getSimulator(params);
+
+        const data: SimulatorReturn = {
```

Comment:
consider extracting a function called `transformSimulatorResponse` - for separation of concerns (fetching data vs. transforming data).

Reply:
this will be a none issue once the backend schema is updated.